org.idempiere.test/src/org/idempiere/test/model/MTestUUTest.java

```diff
@@ -82,10 +82,20 @@ public void testInsertingTestUU() {
 	public void testDeletingTestUU() {
 		Properties ctx = Env.getCtx();
 		String trxName = getTrxName();
-		MTestUU testuu = new MTestUU(ctx, TestRecordInGardenWorld, trxName);
-		testuu.deleteEx(true);
-		MTestUU testuu2 = new MTestUU(ctx, TestRecordInGardenWorld, trxName);
-	    assertFalse(testuu2.get_UUID().equals(TestRecordInGardenWorld));
+
+		// deleting the TestRecordInGardenWorld is creating a deadlock with testReadingUpdatingTestUU in parallel execution
+		// insert a new record
+		MTestUU testuu = new MTestUU(ctx, PO.UUID_NEW_RECORD, trxName);
+		testuu.setName("Test UU record created on JUnit test");
+		testuu.saveEx();
+		testuu.load(trxName);
+	    assertEquals("Test UU record created on JUnit test", testuu.getName());
+	    String uukey = testuu.getTestUU_UU();
+
+		MTestUU testuuToDelete = new MTestUU(ctx, uukey, trxName);
+		testuuToDelete.deleteEx(true);
```

Comment:
before calling delete, should assert `testuuToDelete.get_UUID().equals(uuKey)` is true.